src/battle-animations.ts

```diff
@@ -2708,9 +2708,9 @@ class PokemonSprite extends Sprite {
 		}
 		if (pokemon.volatiles.typechange && pokemon.volatiles.typechange[1]) {
 			let types = pokemon.volatiles.typechange[1].split('/');
-			status += '<img src="' + Dex.resourcePrefix + 'sprites/types/' + encodeURIComponent(types[0]) + '.png" alt="' + types[0] + '" class="pixelated" /> ';
-			if (types[1]) {
-				status += '<img src="' + Dex.resourcePrefix + 'sprites/types/' + encodeURIComponent(types[1]) + '.png" alt="' + types[1] + '" class="pixelated" /> ';
+			for (let i in types) {
+				const type = types[i];
```

Comment:
I'm surprised eslint isn't complaining about this. Oh, we're using an old tslint in the client, maybe that's why. Anyway, using `for...in` on arrays is considered a JS anti-pattern _and_ we're doing polyfills in a way that break it.

```suggestion
			for (const type of types) {
```

Reply:
Oh...I wondered why that felt so convoluted...